spec/acceptance/keystore_spec.rb

```diff
@@ -18,101 +21,104 @@
-          ensure  => file,
-          content => 'puppet',
-        }
-        java_ks { 'puppetca2:keystore':
-          ensure        => latest,
-          certificate   => "#{@temp_dir}ca2.pem",
-          target        => '#{target}',
-          password_file => '#{@temp_dir}password',
-          trustcacerts  => true,
-          path          => #{@resource_path},
-          require       => File['#{@temp_dir}password']
-        }
-      MANIFEST
+      it 'uses a password_file' do
+        pp_two = <<-MANIFEST
+          file { '#{@temp_dir}password':
+            ensure  => file,
+            content => 'puppet',
+          }
+          java_ks { 'puppetca2:keystore':
+            ensure        => latest,
+            certificate   => "#{@temp_dir}ca2.pem",
+            target        => '#{target}',
+            password_file => '#{@temp_dir}password',
+            trustcacerts  => true,
+            path          => #{@resource_path},
+            require       => File['#{@temp_dir}password']
+          }
+        MANIFEST
 
-      idempotent_apply(default, pp_two)
-    end
+        idempotent_apply(pp_two)
+      end
 
-    it 'recreates a keystore if password fails' do
-      pp_three = <<-MANIFEST
+      it 'recreates a keystore if password fails' do
+        pp_three = <<-MANIFEST
 
-        java_ks { 'puppetca:keystore':
-          ensure              => latest,
-          certificate         => "#{@temp_dir}ca.pem",
-          target              => '#{target}',
-          password            => 'pepput',
-          password_fail_reset => true,
-          trustcacerts        => true,
-          path                => #{@resource_path},
-      }
-      MANIFEST
+          java_ks { 'puppetca:keystore':
+            ensure              => latest,
+            certificate         => "#{@temp_dir}ca.pem",
+            target              => '#{target}',
+            password            => 'pepput',
+            password_fail_reset => true,
+            trustcacerts        => true,
+            path                => #{@resource_path},
+        }
+        MANIFEST
 
-      idempotent_apply(default, pp_three)
-    end
+        idempotent_apply(pp_three)
+      end
 
-    it 'verifies the keystore again' do
-      shell("\"#{@keytool_path}keytool\" -list -v -keystore #{target} -storepass pepput") do |r|
-        expect(r.exit_code).to be_zero
-        expectations.each do |expect|
-          expect(r.stdout).to match(expect)
+      it 'verifies the keystore again' do
+        run_shell(keystore_list(target, 'pepput'), expect_failures: true) do |r|
+          expectations.each do |expect|
+            expect(r.stdout).to match(expect)
+          end
+        end
+      end
+
+      it 'cleans up the test artifacts' do
+        run_shell("rm #{target}") do |r|
+          expect(r.exit_code).to be_zero
         end
       end
     end
   end
 
-  unless os[:family] == 'ubuntu' && os[:release].start_with?('18.04')
```

Comment:
This was removed because a comment on a different instance of this check said that this command would not work specifically on Ubuntu 18.04, but in my testing it worked fine. I think the blocker for this has been removed.